fastestimator/trace/io/model_saver.py

```diff
@@ -12,10 +12,31 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 # ==============================================================================
+from typing import Union
 
+import tensorflow as tf
+import torch
+
+from fastestimator.backend.save_model import save_model
 from fastestimator.trace.trace import Trace
+from fastestimator.util import Data
 
 
 class ModelSaver(Trace):
-    def __init__(self, model):
-        pass
+    """save model weights based on epoch frequency during training
+
+    Args:
+        model: model instance
+        save_dir: folder path to save model
+        frequency: model saving frequency in epoch(s). Defaults to 1.
```

Comment:
users never want to save by step frequency? we could perhaps allow str frequencies like "100s" or "2e"

Reply:
I can see the usecase for 100s or 100step.   I guess we will add that in the future if requested, not in this PR tho.

Reply:
k